tools/topology/topology1/sof-adl-nau8825.m4

```diff
@@ -0,0 +1,350 @@
+#
+# Topology for Alderlake with headphone, DMIC, HDMI with/without speaker.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+include(`hda.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Alderlake DSP configuration
+include(`platform/intel/adl.m4')
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+ifdef(`AMP_SSP',`ifdef(`SMART_AMP', , `
```

Comment:
this can be simplified to only use ifdef(SMART_AMP)

Reply:
@plbossart , the nested condition is considering to use for sof-adl-nau8825.tplg.
Because it's without AMP_SSP definition. When no AMP_SSP and not SMART_AMP, it goes to max98360 and rt1019. 
That's why I coded these
ifdef(`AMP_SSP',`ifdef(`SMART_AMP', ,

Reply:
I see. That's really difficult to handle, because if AMP_SSP is not defined, it could be a feature or a bug.

I would recommend using a clearer set of defines

ifdef
NO_AMP -> jack only
else
  AMP_SSP is required
   SMART_AMP is optional
endif

Reply:
Thanks for the advice, @plbossart . I did the modification.